tests/span_tests.cpp

```diff
@@ -483,6 +483,66 @@ SUITE(span_tests)
 #endif
     }
 
+    TEST(from_unique_pointer_construction)
+    {
+        {
+            auto arr = new int[4];
+            auto ptr = std::unique_ptr<int>(arr);
+
+            for (auto i = 0; i < 4; i++)
+                arr[i] = i + 1;
+
+            {
+                span<int> s{arr, 4};
+                CHECK(s.length() == 4 && s.data() == ptr.get());
+                CHECK(s[0] == 1 && s[1] == 2);
+            }
+
+            {
+                span<int> s{std::unique_ptr<int>{nullptr}, 0};
```

Comment:
While I get that it's probably ok in this exact test case, it seems like it might encourage a misunderstanding to construct a `span` from a temporary `unique_ptr`. For any initial value other than `nullptr` that would result in disaster ;-)